MAINTAINERS.md

```diff
@@ -89,31 +89,40 @@ A simple majority vote is required to authorize a SPIRE release at a specific co
+* Raise an issue "Release SPIRE X.Y.Z", and include the release candidate commit hash. Reference the pull request with the updates to the CHANGELOG.
 
 **If this is a major release**, the following steps must be completed before releasing:
-* Review and exercise all examples in spiffe.io and spire-examples repo against the release candidate hash
-* Raise a PR for every example that updates included text and configuration to reflect current state and best practice
-  * Do not merge this PR yet. It will be updated later to use the real version pin rather than the commit hash
-  * If anything unusual is encountered during this process, a comment MUST be left on the release issue describing what was observed
+* Review and exercise all examples in spiffe.io and spire-examples repo against the release candidate hash.
+* Raise a PR for every example that updates included text and configuration to reflect current state and best practice.
+  * Do not merge this PR yet. It will be updated later to use the real version pin rather than the commit hash.
+  * If anything unusual is encountered during this process, a comment MUST be left on the release issue describing what was observed.
 
 The following steps must be completed to perform a release:
-* Cut two annotated tags against the release candidate named `vX.X.X` and `proto/spire/vX.X.X`, where `X.X.X` is the semantic version number of SPIRE
-  * The first line of the annotation should be `X.X.X` followed by the changelog. Refer to previous annotated tags as an example
+* Mark the pull request to update the CHANGELOG as "Ready for review". Make sure that it is updated with the final release date. **At least two approvals from maintainers are required in order to be able to merge it.**
+* If the current state of the master branch has diverged from the candidate commit due to just the CHANGELOG changes, the candidate commit is now the one that includes the updated CHANGELOG and the release can be done from master branch.
+* If the current state of the master branch has diverged from the candidate commit due to other changes than the ones from the CHANGELOG:
+  * If there is not a version branch for this release, create a branch following the guidelines described in [Version branches](#version-branches).
+  * Create a GitHub project named `Release vX.X.X` to identify the PRs that will be cherry-picked. The project should have two statuses to track the progress: one to idenify the PRs to be cherry-picked and one for those that have been merged in the version branch.
+  * Make sure that the version in the branch has been bumped to the version that is being released.
+  * Cherry-pick into the version branch the commits for all the changes that must be included in the release, including the updates in the CHANGELOG and update the corresponding status in the project.
```

Comment:
I feel this step, and the three preceding it, should come as "completed one week prior" steps? That way, we can have predictable release candidate hash well in advance... the project board would generally be more useful, and we'll know how the cherry-picking process went before release time

Reply:
Sounds good!